rundetection/rules/common_rules.py

```diff
@@ -41,3 +49,83 @@ def verify(self, job_request: JobRequest) -> None:
                 "If it is a scatter, contains empty or direct in the title and is assumed to be a scatter "
                 "for an empty can run."
             )
+
+
+class MolSpecStitchRule(Rule[bool]):
+    """
+    Enables Tosca, Osiris, and Iris Run stitching
+    """
+
+    @staticmethod
+    def _is_title_similar(title: str, other_title: str) -> bool:
+        """
+        Compare one run title to another to check for similarity
+        :param title:the first run title
+        :param other_title:the second run title
+        :return: (bool) True if similar False otherwise
+        """
+        logger.info("Comparing titles %s and %s", title, other_title)
+        if title == other_title:
+            return True
+        if title[:-5] == other_title[:-5]:
+            return True
+        if title[0:7] == other_title[0:7] and ("run" in other_title or "run" in title):
+            return True
+        logger.info("Titles not similar, continuing")
+        return False
+
+    def _get_runs_to_stitch(self, run_path: Path, run_number: int, run_title: str, instrument: str) -> list[int]:
+        run_numbers = []
+        while run_path.exists():
+            logger.info("run path exists %s", run_path)
+            if not self._is_title_similar(get_run_title(run_path), run_title):
+                logger.info("titles not similar")
+                break
+            logger.info("titles are similar appending run number %s", run_number)
+            run_numbers.append(run_number)
+            run_number -= 1
+            if instrument.upper() == "TOSCA":
+                next_run_number = f"TSC{run_number}.nxs"
+            else:
+                next_run_number = f"{instrument}{run_number:08d}.nxs"
+            run_path = Path(run_path.parent, next_run_number)
+        logger.info("Run path %s does not exist", run_path)
+        logger.info("Returning run numbers %s", run_numbers)
+        return run_numbers
+
+    def verify(self, job_request: JobRequest) -> None:
+        if not self._value:  # if the stitch rule is set to false, skip
+            return
+
+        logger.info("Checking stitch conditions for %s run %s", job_request.instrument, job_request.filepath)
+        if job_request.instrument.upper() == "OSIRIS":
+            try:
+                if job_request.additional_values["mode"] == "diffraction":
+                    job_request.additional_values["sum_runs"] = False
+                    logger.info("Diffraction run cannot be summed. Continuing")
+                    return
+            except KeyError:
+                pass
```

Comment:
Should this error be logged instead of catching and pass(ing)?

Reply:
It happens in normal function, if mode is not defined which could occur depending on call order, we don't want an uncaught exception. Effectively it's future-proofing now for when we update the OSIRIS code to handle diffractions correctly.